src/averageAges.js

```diff
@@ -37,7 +44,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person =>
+    withChildren
+      ? people.some(child => child.mother === person.name)
+      : person.sex === 'f'
+  );
+
+  const womenAges = women.map(person => person.died - person.born);
+
+  const totalAge = womenAges.reduce((age, average) => age + average);
```

Comment:
You use reduce in every function. It should be added to the `calculateAverageAge` function.